avm/src/vm/opcode.rs

```diff
@@ -2468,7 +2468,37 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       seq[0].1 = current;
     })
   });
-
+  io!("selfrec", |args, mem| {
+    Box::pin(async move {
+      let mut hand_mem = mem.write().await;
+      let mut hm = hand_mem.fork();
+      let (a, b) = hm.addr_to_idxs(args[0]);
+      hm.set_addr(CLOSURE_ARG_MEM_START + 1, a, b);
+      let (a, b) = hm.addr_to_idxs(args[1]);
+      hm.set_addr(CLOSURE_ARG_MEM_START + 2, a, b);
+      let slf = hm.read_fractal(args[0]);
+      let (mut seq, _) = hm.read_either_idxs(slf[0].0, slf[0].1 as usize);
+      let recurse_fn = HandlerFragment::new(slf[1].1, 0);
+      if seq[0].1 < seq[1].1 {
```

Comment:
this is not really part of this PR, but I'm not a big fan of tuples just because they hurt legibility. also what does `slf` stand for?

Reply:
So, the tuples are the core of the memory model now. Rather than add some more memory model function calls, I just dug into it in just this one opcode as a trade-off.

And `slf` is `self` but the Rust compiler wouldn't let me call it that because the `self` variable is a reserved keyword only allowed inside of `impl` blocks for some insane reason.

Reply:
> So, the tuples are the core of the memory model now. Rather than add some more memory model function calls, I just dug into it in just this one opcode as a trade-off.
> 
> And `slf` is `self` but the Rust compiler wouldn't let me call it that because the `self` variable is a reserved keyword only allowed inside of `impl` blocks for some insane reason.

ahh got it!